### PR TITLE
Add Debian9 image and tests.

### PR DESCRIPTION
**Pull request by xingao267**

### CODE REVIEW

**Comment by xingao267** [comment]
cc @nlopezgi

File: debian9/BUILD [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+toolchain_container(
+    name = "builder",
+    base = "@debian_base//image",
+    files = [":mkimage.sh"],
+    packages = [
+        "debootstrap",
+    ],
+)
+
+container_run_and_extract(
```
Comment (nlopezgi) [inline debian9/BUILD]: add brief doc above about what this does (does this also need a TODO to push this file to GCS bucket)?

File: debian9/BUILD [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+container_image(
+    name = "image",
+    cmd = "/bin/bash",
+    env = DEBIAN_ENV,
+    tars = [
+        # The overlay tar has to come first to actuall overwrite existing files.
+        ":overlay.tar",
+        # TODO(xingao): Switch to use checked-in file.
```
Comment (nlopezgi) [inline debian9/BUILD]: any particular reason why you're not doing this as part of this PR (just curious)?

File: debian9/WORKSPACE [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+)
+
+container_repositories()
+
+load(
+    "@io_bazel_rules_docker//container:container.bzl",
+    "container_pull",
+)
+
+# Pull existing Debian base, only used to create builder image to debootstrap.
```
Comment (nlopezgi) [inline debian9/WORKSPACE]: Why is this needed (is it just because we need some existing image to bootstrap)?
Do we need a TODO to switch to using some other strategy (or will we eventually use the images produced by our process to bootstrap producing other images)?

File: debian9/deps_spec.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+revisionsFilePath: "debian9/revisions.bzl"
+
+gcsDeps:
```
Comment (nlopezgi) [inline debian9/deps_spec.yaml]: Which other dependencies do we expect to track here, if any (can we list them out as TODOs)?

File: debian9/BUILD (hunk 1 above)
Comment (xingao267) [reply debian9/BUILD]: Done. I've pushed it to the GCS bucket.

File: debian9/BUILD (hunk 2 above)
Comment (xingao267) [reply debian9/BUILD]: I didn't do this because I want this strategy/modification to the existing way to generate the image to be approved first before I upload any file to the GCS bucket. Since it's approved, I'm gonna do it and modify this PR accordingly.

File: debian9/WORKSPACE (hunk 3 above)
Comment (xingao267) [reply debian9/WORKSPACE]: Discussed offline. This strategy is fine. We'll think of strategy how to update the container pin here. But it shouldn't matter too much where we run debootstrap.

File: debian9/deps_spec.yaml (hunk 4 above)
Comment (xingao267) [reply debian9/deps_spec.yaml]: This is all.
